src/ws-api.ts

```diff
@@ -35,127 +30,14 @@ export class WsApi {
-            JsonRpcErrorCodes.BadRequest,
-            'request payload required.',
-          );
-
-          const responseBuffer = WsApi.jsonRpcResponseToBuffer(jsonRpcResponse);
-          return socket.send(responseBuffer);
-        }
-
-        dwnRequest = JSON.parse(dwnRequest);
-      } catch (e) {
-        const jsonRpcResponse = createJsonRpcErrorResponse(
-          uuidv4(),
-          JsonRpcErrorCodes.BadRequest,
-          e.message,
-        );
-
-        const responseBuffer = WsApi.jsonRpcResponseToBuffer(jsonRpcResponse);
-        return socket.send(responseBuffer);
-      }
-
-      // Check whether data was provided in the request
-      const { encodedData } = dwnRequest.params;
-      const requestDataStream = encodedData
-        ? DataStream.fromBytes(base64url.baseDecode(encodedData))
-        : undefined;
-
-      const requestContext: RequestContext = {
-        dwn,
-        transport: 'ws',
-        dataStream: requestDataStream,
-      };
-      const { jsonRpcResponse } = await jsonRpcApi.handle(
-        dwnRequest,
-        requestContext,
-      );
-
-      if (jsonRpcResponse.error) {
-        requestCounter.inc({ method: dwnRequest.method, error: 1 });
-      } else {
-        requestCounter.inc({
-          method: dwnRequest.method,
-          status: jsonRpcResponse?.result?.reply?.status?.code || 0,
-        });
-      }
-
-      const responseBuffer = WsApi.jsonRpcResponseToBuffer(jsonRpcResponse);
-      return socket.send(responseBuffer);
-    });
-  }
-
-  /**
-   * This handler returns an interval to ping clients' socket every 30s
-   * if a pong hasn't received from a socket by the next ping, the server will terminate the socket connection.
-   */
-  #setupHeartbeat(): NodeJS.Timer {
-    // Sometimes connections between client <-> server can get borked in such a way that
-    // leaves both unaware of the borkage. ping messages can be used as a means to verify
-    // that the remote endpoint is still responsive. Server will ping each socket every 30s
-    // if a pong hasn't received from a socket by the next ping, the server will terminate
-    // the socket connection
-    return setInterval(() => {
-      this.#wsServer.clients.forEach(function (socket) {
-        if (socket[SOCKET_ISALIVE_SYMBOL] === false) {
-          return socket.terminate();
-        }
-
-        socket[SOCKET_ISALIVE_SYMBOL] = false;
-        socket.ping();
-      });
-    }, HEARTBEAT_INTERVAL);
-  }
-
   /**
    * Handler for starting a WebSocket.
    * Sets listeners for `connection`, `close` events.
    * It clears `heartbeatInterval` when a `close` event is made.
    */
   #setupWebSocket(): void {
-    this.#wsServer.on('connection', this.#handleConnection.bind(this));
-
-    const heartbeatInterval = this.#setupHeartbeat();
-
-    this.#wsServer.on('close', function close() {
-      clearInterval(heartbeatInterval);
-    });
+    this.#wsServer.on('connection', this.#connections.connect.bind(this));
```

Comment:
Sanity: should `bind(this)` actually be `bind(this.#connection)`?

I'd actually suggest avoid using `bing()` at all, I always find the extra level of indirection confusing (case in point). Would this work:
```suggestion
    this.#wsServer.on('connection', (socket, request) => this.#connections.connect(socket, request));
```

Same goes for `on('close'...`

Reply:
Good catch! I also hate using .bind() typically for this reason, but it's a bit "cleaner" and i see being used as the standard in most places. But I'm definitely ok with moving away from that!

Reply:
Done